exploration/0001-builtin-registry-capabilities.md

```diff
@@ -0,0 +1,52 @@
+# Capabilities for built-ins
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Contributors</dt>
+		<dd>@ryzokuken</dd>
+		<dt>First proposed</dt>
+		<dd>2023-08-22</dd>
+		<dt>Pull Request</dt>
+		<dd>#457</dd>
+	</dl>
+</details>
+
+## Objective
+
+Document the capabilities that the set of built-in functions would expose to the user in order to ensure it can be designed in the most minimalistic way that would still cater to the most pressing requirements, partly inspired by ICU4X anbd JavaScript's Intl API and backwards-compatible with MF1 in ICU4J.
+
+## Background
+
+Users have the ability to specify and call arbitrary custom functions in their messages thanks to the function registry.
+That said, there are a number of common i18n operations that are quite universal and there's a lot of value in standardizing the usage of these common operations across the ecosystem.
+This document aims to list down the basic capabilities for developers that such a set of standard built-in functions MUST provide.
+
+## Use-Cases
+
+From formatting date and time values to performing selection based on plural rules, the use cases for a minimal set of common tools standardized across implementations are virtually everywhere. While a number of applications might require more complex functions with custom logic, a decently designed set of built-in functions could significantly lower the barrier of entry as well as the need to write many custom functions.
+
+## Requirements
+
+TBD
+
+## Constraints
+
+TBD
+
+## Proposed Design
+
+All conforming implementations of this specification MUST provide a set of <dfn>built-in functions</dfn>, which allows the developers to perform the following operations:
+
+* Formatting the following types of data in a locale-sensitive manner:
+    * Dates and times
+    * Numbers
+    * Lists
+* Selecting between pattern strings in a locale-sensitive manner based on:
+	* String equality
+    * Plural rules
+    * Ordinality
```

Comment:
Isn't ordinality a subset of plural rules? I know that MF1 has separately `plural` and `selectordinal`, but wouldn't it make sense for our plural selector to include something like `type=ordinal` as an option instaed?

Reply:
In my basic understanding of linguistics, there's not a whole lot of overlap between the two concepts. We could obviously decide to design the API either way, I have no strong opinions but we should err on the side of not confusing the user between two otherwise distinct linguistic concepts.

Reply:
@eemeli Yes, you're correct and we might cover it using an option. But it is good to ensure we remember it on the checklist of things we need to build in.